[MRG+2] Fix Support sample_weight in label_ranking_average_precision_score

Jopepato opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
Fixes #10845 and resolves #10840 

#### What does this implement/fix? Explain your changes.
It adds the sample_weight parameter in the function label_ranking_average_precision_score for metrics

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Jopepato committed:
Adding sample_weight for label_ranking_average_precision_score and documenting it

jnothman commented on sklearn/metrics/ranking.py:
```diff
@@ -712,7 +716,7 @@ def label_ranking_average_precision_score(y_true, y_score):
         out += (L / rank).mean()
 
-    return out / n_samples
+    return np.average(out / n_samples, weights=sample_weight)
```
For this to work, out needs to be an array. Alternatively you can do the weighted averaging yourself by multiplying each contribution to out and dividing by the sum of weights

jnothman reviewed (commented):
Please update test_common to indicate weights are supported

Jopepato committed:
Fixed array problem

Jopepato commented:
@jnothman I have fixed the array problem with out, but I don't know where to indicate that weights are supported in test_common

jnothman commented:
It took a bit of investigation, but this is it:
label_ranking_average_precision_score is incorrectly missing from
THRESHOLDED_MULTILABEL_METRICS.

On 29 March 2018 at 23:32, Jose Perez-Parras Toledano <
notifications@github.com> wrote:

> @jnothman <https://github.com/jnothman> I have fixed the array problem
> with out, but I don't know where to indicate that weights are supported in
> test_common
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/10860#issuecomment-377220556>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz6-60LwNle4qxNrkuWozCnjjNd25Wks5tjNRxgaJpZM4S3wY->
> .
>

Jopepato committed:
Added label_ranking_average_precision_score to the tests

Jopepato committed:
Added the comprobation for the sample_weights when its included

Jopepato committed:
Deleted whitespace which caused travis to fail

Jopepato commented:
@jnothman I finally added the function to the tests (as you said) and implemented the sample_weights in a correct way so it's working now
Sorry for the delay, but I learnt how the code was organized looking through the tests ^^

jnothman commented on sklearn/metrics/ranking.py:
```diff
@@ -710,9 +714,19 @@ def label_ranking_average_precision_score(y_true, y_score):
+            out += aux
+        else:
+            aux = aux * sample_weight[i]
+            out += aux
```
pull this out of the if so that you only have a single `if sample_weight is not None` case.

jnothman commented:
Otherwise LGTM.

Please add an entry to the Enhancements  change log at `doc/whats_new/v0.20.rst`. Like the other entries there, please reference this pull request with `:issue:` and credit yourself (and other contributors if applicable) with `:user:`

Commit:
Optimized code and added issue and user to docs

Commit:
Making travis happy again (deleting whitespaces)

Jopepato commented:
@jnothman I changed the code you told me and added the issue in the docs, thanks ^^

qinhanmin2014 commented on doc/whats_new/v0.20.rst:
```diff
@@ -195,6 +195,9 @@ Metrics
 - :func:`metrics.roc_auc_score` now supports binary ``y_true`` other than
   ``{0, 1}`` or ``{-1, 1}``. :issue:`9828` by :user:`Hanmin Qin <qinhanmin2014>`.
 
+- :func:`label_ranking_average_precision_score` now supports vector ``sample_weight``
```
``metrics.label_ranking_average_precision_score`` and a ``.`` at the end.

qinhanmin2014 commented on sklearn/metrics/ranking.py:
```diff
@@ -682,6 +685,7 @@ def label_ranking_average_precision_score(y_true, y_score):
     check_consistent_length(y_true, y_score)
```
This is duplicate. Please remove this line and move ``check_consistent_length(y_true, y_score, sample_weight)`` here.

qinhanmin2014 reviewed (approved):
LGTM @Jopepato please use keywords to create links to relevant issues so that they can be closed automatically. See https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/.

Commit:
Fixed sentence in the docs and eliminated duplicated check in metrics

Jopepato commented:
@qinhanmin2014 I've changed the description of the PR for closing the issues, thanks for the tutorial I didn't know about it ^^
And also added the changes you told me

qinhanmin2014 reviewed (approved):
LGTM Thanks @Jopepato :)
